File: 36. Valid Sudoku/Solution.py
class Solution:
    def isValidSudoku(self, board):
        """
        :type board: List[List[str]]
        :rtype: bool
        """
        size = len(board)
        l = [0 for x in range(size + 1)]
        for i in range(size):
            for u in range(size + 1):
                l[u] = 0
            for j in range(size):
                if board[i][j] == '.':
                    continue
                if l[int(board[i][j])] == 1:
                    return False
                else:
                    l[int(board[i][j])] += 1
        for i in range(size):
            for u in range(size + 1):
                l[u] = 0
            for j in range(size):
                if board[j][i] == '.':
                    continue
                if l[int(board[j][i])] == 1:
                    return False
                else:
                    l[int(board[j][i])] += 1
        for i in range(0, size - 1, 3):
            for j in range(0, size - 1, 3):
                for u in range(size + 1):
                    l[u] = 0
                for m in range(3):
                    for n in range(3):
                        if board[i + m][j + n] == '.':
                            continue
                        int(board[i + m][j + n])
                        if l[int(board[i + m][j + n])] == 1:
                            return False
                        else:
                            l[int(board[i + m][j + n])] += 1
        return True

File: 36. Valid Sudoku/Solution.cpp
class Solution
{
  public:
    bool isValidSudoku(vector<vector<char>> &board)
    {
        int size = board.size();
        vector<int> count;

        for (int i = 0; i < size; i++)
        {
            count.assign(size, 0);
            for (int j = 0; j < size; j++)
            {
                if (board[i][j] != '.')
                {
                    int pos = board[i][j] - '1';
                    if (count[pos] != 0)
                        return false;
                    else
                        count[pos]++;
                }
            }
        }

        for (int j = 0; j < size; j++)
        {
            count.assign(9, 0);
            for (int i = 0; i < size; i++)
            {
                if (board[i][j] != '.')
                {
                    int pos = board[i][j] - '1';
                    if (count[pos] != 0)
                        return false;
                    else
                        count[pos]++;
                }
            }
        }

        for (int i = 0; i < size; i += 3)
        {
            for (int j = 0; j < size; j += 3)
            {
                count.assign(9, 0);
                for (int row = i; row < i + 3; row++)
                    for (int col = j; col < j + 3; col++)
                    {
                        if (board[row][col] != '.')
                        {
                            int pos = board[row][col] - '1';
                            if (count[pos] != 0)
                                return false;
                            else
                                count[pos]++;
                        }
                    }
            }
        }

        return true;
    }
};